packages/core/src/api/apis/definitions/auth.ts

```diff
@@ -0,0 +1,149 @@
+   * will be prompted to log in. The returned promise will not resolve until the user has
+   * successfully logged in. The returned promise can be rejected, but only if the user rejects the login request.
+   */
+  getAccessToken(scope?: OAuthScopeLike): Promise<string>;
+
+  /**
+   * Log out the user's session. This will reload the page.
+   */
+  logout(): Promise<void>;
+};
+
+export type IdTokenOptions = {
+  // If this is set to true, the user will not be prompted to log in,
+  // and an empty id token will be returned if there is no existing session.
+  //
+  // This can be used to perform a check whether the user is logged in, or if you don't
+  // want to force a user to be logged in, but provide functionality if they already are.
+  optional?: boolean;
+
+  // If this is set to true, the request will bypass the regular oauth login modal
+  // and open the login popup directly.
+  //
+  // The method must be called synchronously from a user action for this to work in all browsers.
+  instantPopup?: boolean;
+};
+
+/**
+ * This API provides access to OpenID Connect credentials. It lets you request ID tokens,
+ * which can be passed to backend services to prove the user's identity.
+ */
+export type OpenIdConnectApi = {
+  /**
+   * Requests an OpenID Connect ID Token.
+   *
+   * This method is cheap and should be called each time an ID token is used. Do not for example
+   * store the id token in React component state, as that could cause the token to expire. Instead
+   * fetch a new id token for each request.
+   *
+   * If the user has not yet logged in to Google inside Backstage, the user will be prompted
+   * to log in. The returned promise will not resolve until the user has successfully logged in.
+   * The returned promise can be rejected, but only if the user rejects the login request.
+   */
+  getIdToken(options?: IdTokenOptions): Promise<string>;
+
+  /**
+   * Log out the user's session. This will reload the page.
+   */
+  logout(): Promise<void>;
+};
+
+/**
+ * Provides authentication towards Google APIs and identities.
+ *
+ * See https://developers.google.com/identity/protocols/googlescopes for a full list of supported scopes.
+ *
+ * Note that the ID token payload is only guaranteed to contain the user's numerical Google ID,
+ * email and expiration information. Do not rely on any other fields, as they might not be present.
+ */
+export const googleAuthApiRef = new ApiRef<OAuthApi & OpenIdConnectApi>({
+  id: 'core.auth.google',
+  description: 'Provides authentication towards Google APIs and identities',
+});
+
+/**
+ * Provides authentication towards Github APIs.
+ *
+ * See https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/
+ * for a full list of supported scopes.
+ */
+export const githubAuthApiRef = new ApiRef<OAuthApi>({
+  id: 'core.auth.github',
+  description: 'Provides authentication towards Github APIs',
+});
```

Comment:
How will this work of you want to support both github and an internal GHE?

Reply:
Current idea is that plugins depend on this ref, and the app can choose whether GHE or GitHub is the default one, as in the one registered as a part of app. Then we could either allow per-plugin overrides of the implementation, and/or the plugins that want to use the other provider will have to do that explicitly through a different api ref.